src/main/java/Duke/TaskList.java

```diff
@@ -0,0 +1,193 @@
+        }
+
+    }
+
+    public void addTask(Task task, boolean fromdata){
+        this.list.add(task);
+        this.curSize++;
+        String item = "\n" +
+        "   ____________________________________________________________\n" +
+        "   Got it. I've added this task: \n" + 
+        "       " + task +  "\n" +
+        "   Now you have " + curSize + " tasks in the list.\n" +
+        "   ____________________________________________________________";
+        if (!fromdata){
+            System.out.println(item);
+        }
+    }
+
+    public void done(int num){
+        Task task = this.list.get(num  - 1);
+        task.doneTask();
+        this.list.set(num  - 1, task);
+        String item = "\n" +
+        "   ____________________________________________________________\n" +
+        "   nice Nice! I've marked this task as done: \n" +
+        "       " + task.toString() + "\n" + 
+        "   ____________________________________________________________";
+        System.out.println(item);
+
+    }
+
+
+
+    public void actionHalder(Parser actionList, boolean done, boolean fromData) throws InputNotValidError{
+        boolean validCommand;
+        try{
+            validCommand = actionList.isValid();
+        }
+        catch(InputNotValidError e){
+            String error = 
+            "   ____________________________________________________________\n" +
+            "   " + e.getMessage() + '\n' +
+            "   ____________________________________________________________";
+            System.out.println(error);
+            return;
+        }
+        if (validCommand){
+            Action actionName = actionList.getAction();
+
+            switch(actionName){
+                case LIST:
+                    this.printList();
+                    break;
+                case DONE:
+                    int num = Integer.parseInt(actionList.getActionList());
+                    this.done(num);
+                    break;                
+                case TODO:
+                    this.addToDo(actionList.getActionList(), done, fromData);
+                    break;
+                case DEADLINE:
+                    this.addDeadline(actionList.getActionList(), done, fromData);
+                    break;
+                case EVENT:
+                    this.addEvent(actionList.getActionList(), done, fromData);
+                    break;
+                case DELETE:
+                    int num2 = Integer.parseInt(actionList.getActionList());
+                    this.delete(num2);
+                    break;
+                case FIND:
+                    this.find(actionList.getActionList());
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+    
+    public void addToDo(String action, boolean done, boolean fromdata){
+        ToDo newTask = new ToDo(action, done, "T");
+        this.addTask(newTask, fromdata);
+    }
+
```

Comment:
Perhaps you might want to consider following camelCase standard in the naming of your method arguments too.